tests/stream_test_helper.c

```diff
@@ -116,6 +116,14 @@ static int s_on_body(struct aws_http_stream *stream, const struct aws_byte_curso
     ASSERT_SUCCESS(aws_byte_buf_append_dynamic(&tester->response_body, data));
     return AWS_OP_SUCCESS;
 }
+static void s_on_metrics(
+    struct aws_http_stream *stream,
+    const struct aws_http_stream_metrics *metrics,
+    void *user_data) {
+    (void)stream;
+    struct client_stream_tester *tester = user_data;
+    tester->metrics = *metrics;
+}
```

Comment:
trivial: assert that all values are positive?

Reply:
we cannot really do that, because we have tests that the stream was never sent, so data can be unavailable

Reply:
We can add a separate test for metrics and maybe a test where data is unavailable.

Reply:
You could still AWS_FATAL_ASSERT some things:
- that stream_id is set
- if _duration is set, then _end is set
- if _end is set, then _begin is set